src/validation_workflow/validation.py

```diff
@@ -51,6 +53,27 @@ def check_for_security_plugin(self, work_dir: str) -> bool:
         path = os.path.exists(os.path.join(work_dir, "plugins", "opensearch-security"))
         return path
 
+    def install_native_plugin(self, path: str) -> None:
+        self.native_plugins_list = self.get_native_plugin_list(path)
+        for native_plugin in self.native_plugins_list:
+            execute('.' + os.sep + f'opensearch-plugin install --batch {native_plugin}', os.path.join(path, "bin"))
+
+    def get_native_plugin_list(self, workdir: str) -> list:
+        bundle_manifest = BundleManifest.from_path(os.path.join(workdir, "manifest.yml"))
+        commit_id = bundle_manifest.components["OpenSearch"].commit_id
+        plugin_url = f"https://api.github.com/repos/opensearch-project/OpenSearch/contents/plugins?ref={commit_id}"
+        api_response = requests.get(plugin_url)
+        if api_response.status_code == 200:
+            response = api_response.json()
+            installed_plugins_list = os.listdir(os.path.join(workdir, "plugins"))
+            plugin_list = [i["name"] for i in response if i["name"] not in installed_plugins_list]
+            plugin_list.remove("examples")
+            plugin_list.remove("build.gradle")
+            plugin_list.remove("identity-shiro")  # Assuming security plugin enabled in the artifacts
```

Comment:
@Divyaasm can you add some more details like why 2 identity plugins cannot be installed.